mleap-tensorflow/src/test/scala/ml/combust/mleap/tensorflow/converter/MleapConverterSpec.scala

```diff
@@ -0,0 +1,155 @@
+      assertSameTensor(ml_tensor, ml_tensor_converted)
+      // make sure same indices has same value between tensorflow tensor and mleap tensor
+      assert(
+        toIndices(shape).map(indices => tf_tensor_converted.getObject(indices.map(_.toLong): _*))
+          sameElements toIndices(shape).map(indices => ml_tensor(indices: _*))
+      )
+
+    }
+
+    it("double nd dense tensors") {
+      val array = Array.fill(size) {
+        random.nextDouble()
+      }
+      val ml_tensor = Tensor.create(array, dimensions = shape)
+      val tf_tensor_converted = MleapConverter.convert(ml_tensor, TensorType.Double()).asInstanceOf[TFloat64]
+      val ml_tensor_converted = TensorflowConverter.convert(tf_tensor_converted, TensorType.Double())
+      assertSameTensor(ml_tensor, ml_tensor_converted)
+      // make sure same indices has same value between tensorflow tensor and mleap tensor
+      assert(
+        toIndices(shape).map(indices => tf_tensor_converted.getObject(indices.map(_.toLong): _*))
+          sameElements toIndices(shape).map(indices => ml_tensor(indices: _*))
+      )
+    }
+    it("byte nd dense tensors") {
+      val array = Array.fill[Byte](size) {
+        0
+      }
+      random.nextBytes(array)
+      val ml_tensor = Tensor.create(array, dimensions = shape)
+      val tf_tensor_converted = MleapConverter.convert(ml_tensor, TensorType.Byte()).asInstanceOf[TUint8]
+      val ml_tensor_converted = TensorflowConverter.convert(tf_tensor_converted, TensorType.Byte())
+      assertSameTensor(ml_tensor, ml_tensor_converted)
+      // make sure same indices has same value between tensorflow tensor and mleap tensor
+      assert(
+        toIndices(shape).map(indices => tf_tensor_converted.getObject(indices.map(_.toLong): _*))
+          sameElements toIndices(shape).map(indices => ml_tensor(indices: _*))
+      )
+    }
+
+    it("byte string nd dense tensors") {
+      val array = Array.fill[ByteString](size) {
+        val bytes = Array.fill[Byte](100) {
+          'a'
+        }
+        random.nextBytes(bytes)
+        ByteString(bytes)
+      }
+
+      val ml_tensor = Tensor.create(array, dimensions = shape)
+      val tf_tensor_converted = MleapConverter.convert(ml_tensor, TensorType.ByteString()).asInstanceOf[TString]
+      val ml_tensor_converted = TensorflowConverter.convert(tf_tensor_converted, TensorType.ByteString()).asInstanceOf[Tensor[ByteString]]
+      assertSameTensor(ml_tensor, ml_tensor_converted)
+      // make sure same indices has same value between tensorflow tensor and mleap tensor
+      val expected = toIndices(shape).map(indices => ml_tensor(indices: _*).bytes)
+      val actual = toIndices(shape).map(indices => tf_tensor_converted.asBytes().getObject(indices.map(_.toLong): _*))
+      assert(expected.zip(actual).forall({ case (lhs, rhs) => lhs sameElements rhs }))
+    }
+
+    it("string nd dense tensors") {
```

Comment:
We seem to test converting Dense tensors only... can we test Sparse as well?

Reply:
We convert every Mleap Tensor to DenseTensor at the beginning of `MleapConverter.convert`.
But this bring me to a question.
Currently for MLeap SparseTensor, we call `toDense` in `MleapConverter.convert` to covert it to DenseTensor before convert it to Tensorflow Tensor.
This approach will introduce memory problem if the Sparse Tensor is really sparse.
Tensorflow has SparseTensor concept as well.  I am thinking of convert to Tensorflow SparseTensor if it's a MLeap SparseTensor.
@jsleight @ancasarb

Reply:
Respecting the sparse vs dense makes sense to me.  Exactly as you mentioned, could be very memory inefficient as is.

Reply:
Since Tensorflow SparseTensor is not a child of Tensor,  Let me check what we can make the most use of it.

Reply:
I open and issue to track Sparse Tensor.
https://github.com/combust/mleap/issues/764